contracts/mock/MockNativeMetaTransaction.sol

```diff
@@ -6,9 +6,13 @@ pragma solidity 0.8.9;
 import { MockEIP712Base } from "./MockEIP712Base.sol";
 
 contract MockNativeMetaTransaction is MockEIP712Base {
+    constructor(string memory name, string memory version) {
+        _initializeEIP712(name, version);
+    }
+
     bytes32 private constant META_TRANSACTION_TYPEHASH =
-        keccak256(bytes("MetaTransaction(uint256 nonce,address from,bytes functionSignature)"));
-    event MetaTransactionExecuted(address userAddress, address payable relayerAddress, bytes functionSignature);
+        keccak256(bytes("MetaTransaction(uint256 nonce,address from,address to,bytes functionSignature)"));
```

Comment:
The initial implementation of MockNativeMetaTransaction is compliant with all Child Token on polygon.
Example: Boson Token (proxy: 0x9b3b0703d392321ad24338ff1f846650437a43c9) --> impl: https://polygonscan.com/address/0xe459bb24105f5263be2b99bebd213d7e12fc399c#code (line 1336).
So changing this implementation (and removing the inheritance for Foreign20 is breaking core-components tests to insure meta transactions can be used on Polygon tokens, for instance for approval methods).
If the BosonVoucher contract needs another meta-tx implementation, I would suggest to implement another base class (for instance "VoucherMetaTransactionForwarder" ?)

Reply:
See my comment above